torchci/scripts/check_alerts.py

```diff
@@ -158,10 +167,20 @@ def get_most_recent_failure_chain(self) -> List[Any]:
         return failures
 
     def should_alert(self) -> bool:
+        # Group jobs by their failures. The length of the failure chain is used
+        # to raise the alert, so we can do a simple tweak here to use the length
+        # of the longest unique chain
+        unique_failures = self.get_unique_failures(self.failure_chain)
+
         return (
             self.current_status is not None
-            and self.current_status["conclusion"] != "success"
-            and len(self.failure_chain) >= FAILURE_CHAIN_THRESHOLD
+            and self.current_status["conclusion"] != SUCCESS
+            and any(
+                [
+                    len(failure_chain) >= FAILURE_CHAIN_THRESHOLD
+                    for failure_chain in unique_failures.values()
+                ]
```

Comment:
No need to extra braces in any, isn't it? (I.e. `all(x>3 for x in range(5))` feels like a valid statement)
```suggestion
                len(failure_chain) >= FAILURE_CHAIN_THRESHOLD
                for failure_chain in unique_failures.values()
```

Reply:
Oh, you're right. TIL.